### PR TITLE
[Shop] Currency switcher

### PR DESCRIPTION
**Pull request by mpysiak**
| Q               | A
|-----------------|-----
| Branch?         | bootstrap-shop
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | n/a
| License         | MIT

Made currency switcher twig component, enabled currency scenarios. Refactored CardComponent - removed duplicated PriceComponents

### CODE REVIEW

File: features/shop/currency/switching_current_currency.feature [hunk 1]
```diff
@@ -8,22 +8,22 @@ Feature: Switching the current currency
...
         Then I should shop using the "EUR" currency
 
-    @no-api @todo-ui
+    @no-api @ui
     Scenario: Showing available currencies
         When I browse that channel
         Then I should be able to shop using the "USD" currency
 
-    @api @todo-ui
+    @api @no-ui
```
Comment (Wojdylak) [inline features/shop/currency/switching_current_currency.feature]: Why is this `@no-ui`?

File: src/Sylius/Bundle/ShopBundle/Twig/Component/Common/CurrencySwitcherComponent.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+use Sylius\Component\Currency\Model\CurrencyInterface;
+use Sylius\TwigHooks\Twig\Component\HookableComponentTrait;
+use Symfony\UX\LiveComponent\DefaultActionTrait;
+use Symfony\UX\TwigComponent\Attribute\AsTwigComponent;
+use Symfony\UX\TwigComponent\Attribute\ExposeInTemplate;
+
+#[AsTwigComponent]
+class CurrencySwitcherComponent
+{
+    use DefaultActionTrait;
```
Comment (Wojdylak) [inline src/Sylius/Bundle/ShopBundle/Twig/Component/Common/CurrencySwitcherComponent.php]:
```suggestion
```

File: features/shop/currency/switching_current_currency.feature (hunk 1 above)
Comment (mpysiak) [reply features/shop/currency/switching_current_currency.feature]: Added  🙌

File: src/Sylius/Bundle/ShopBundle/Twig/Component/Product/PriceComponent.php [hunk 3]
```diff
@@ -30,50 +33,58 @@ class PriceComponent
...
     public ProductVariant $variant;
 
+    #[ExposeInTemplate]
+    public ?string $price = null;
+
+    #[ExposeInTemplate(name: 'original_price')]
+    public ?string $originalPrice = null;
+
+    #[ExposeInTemplate(name: 'has_discount')]
+    public bool $hasDiscount = false;
```
Comment (Rafikooo) [inline src/Sylius/Bundle/ShopBundle/Twig/Component/Product/PriceComponent.php:43]: Why is there a missing name declaration here? Are the following defined to enforce snake case?
```suggestion
    #[ExposeInTemplate(name: 'price')]
    public ?string $price = null;

    #[ExposeInTemplate(name: 'original_price')]
    public ?string $originalPrice = null;
    ...
```

**Review by Rafikooo — APPROVED** [review]

**Pull request merged**